Makefile

```diff
@@ -238,54 +101,71 @@ update_devworkspace_api:
+
+### install: Install CRDs into a cluster
+install: manifests _kustomize _create_namespace
+	$(KUSTOMIZE) build config/crd | kubectl apply -f -
+
+#### uninstall: Uninstall CRDs from a cluster
+uninstall: manifests _kustomize
+	$(KUSTOMIZE) build config/crd | kubectl delete -f -
+
+### deploy: Deploy controller in the configured Kubernetes cluster in ~/.kube/config
+deploy: _kustomize _create_namespace deploy_registry
+	mv config/devel/config.properties config/devel/config.properties.bak
+	mv config/devel/manager_image_patch.yaml config/devel/manager_image_patch.yaml.bak
+	envsubst < config/devel/config.properties.bak > config/devel/config.properties
+	envsubst < config/devel/manager_image_patch.yaml.bak > config/devel/manager_image_patch.yaml
+	$(KUSTOMIZE) build config/devel | kubectl apply -f - || true
+	mv config/devel/config.properties.bak config/devel/config.properties
+	mv config/devel/manager_image_patch.yaml.bak config/devel/manager_image_patch.yaml
+
+### restart: Restart devworkspace-controller deployment
+restart:
+	kubectl rollout restart -n $(NAMESPACE) deployment/devworkspace-controller-manager
+
+### uninstall_controller: Remove controller resources from the cluster
+uninstall_controller: _kustomize
+	kustomize build config/devel | kubectl delete --ignore-not-found -f -
+
+### deploy_registry: Deploy plugin registry
+deploy_registry: _create_namespace
+	kubectl apply -f config/registry/local -n $(NAMESPACE)
+ifeq ($(PLATFORM),kubernetes)
+	envsubst < config/registry/local/k8s/ingress.yaml | kubectl apply -n $(NAMESPACE) -f -
 else
-	$(error operator-sdk $(OPERATOR_SDK_VERSION) is expected to be used during CRDs and k8s objects generating while $(shell operator-sdk version | cut -d , -f 1 | cut -d : -f 2 | cut -d \" -f 2) found)
-endif
-
-### start_local: start local instance of controller using operator-sdk
-start_local: _bump_kubeconfig _generate_related_images_env _login_with_devworkspace_sa
-	@source $(RELATED_IMAGES_FILE)
-ifeq ($(WEBHOOK_ENABLED),true)
-	#in cluster mode it comes from Deployment env var
-	export RELATED_IMAGE_devworkspace_webhook_server=$(IMG)
-	#in cluster mode it comes from configured SA propogated via env var
-	export CONTROLLER_SERVICE_ACCOUNT_NAME=devworkspace-controller
-	export KUBECONFIG=$(BUMPED_KUBECONFIG)
-endif
-	operator-sdk run --local --watch-namespace $(NAMESPACE) 2>&1 | grep --color=always -E '"msg":"[^"]*"|$$'
-
-### start_local_debug: start local instance of controller with debugging enabled
-start_local_debug: _bump_kubeconfig _generate_related_images_env _login_with_devworkspace_sa
-	@source $(RELATED_IMAGES_FILE)
-ifeq ($(WEBHOOK_ENABLED),true)
-	#in cluster mode it comes from Deployment env var
-	export RELATED_IMAGE_devworkspace_webhook_server=$(IMG)
-	#in cluster mode it comes from configured SA propogated via env var
-	export CONTROLLER_SERVICE_ACCOUNT_NAME=devworkspace-controller
-	export KUBECONFIG=$(BUMPED_KUBECONFIG)
+	kubectl apply -f config/registry/local/os -n $(NAMESPACE)
```

Comment:
that's strange but it fails for me on crc
```kubectl apply -f config/registry/local/os -n devworkspace-controller
error: unable to recognize "config/registry/local/os/route.yaml": no matches for kind "Route" in version "v1"
Makefile:154: recipe for target 'deploy_registry' failed
make: *** [deploy_registry] Error 1
```
> ~/projects/devworkspace-operator update-sdk-1.x *3 !1 > crc version                                                                                              > % 14:35:22
> CodeReady Containers version: 1.15.0+e317bed
> OpenShift version: 4.5.7 (embedded in binary)

After changing Route api version to `apiVersion: route.openshift.io/v1 to fail in another place

Reply:
Hm I'll re-test this today. The files are copied over from the current repo IIRC so this shouldn't be happening.

Reply:
Interesting; `oc` recognizes `v1/Route` through some strange magic but `kubectl` does not. I've updated `route.yaml` to use the full apiVersion (`route.openshift.io/v1`).